### PR TITLE
feat: update L1 origin block hash handling and add full L1 origin block hash field

### PR DESCRIPTION
**Pull request by luis-herasme**
### Checklist

- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
If the full block hash cannot be found, this change avoids adding a link to the block hash field, as the URL would be incorrect.

### Related Issue
#630

### CODE REVIEW

**Review by PJColombo — CHANGES_REQUESTED** [review]
Left a comment.

Additionally, the [autocompleteBlockHash](https://github.com/Blobscan/blobscan/blob/88055e24a2dbbe151ce69726fc410e9a00a07d9a/packages/api/src/utils/autocompleteBlockHash.ts#L8C67-L9C1) function is not working correctly.

We are filtering hashes that start with the `partialHash` [here](https://github.com/Blobscan/blobscan/blob/88055e24a2dbbe151ce69726fc410e9a00a07d9a/packages/api/src/utils/autocompleteBlockHash.ts#L12). However, this condition will never hold because the partial hash is not prefixed with `0x`, whereas the block hashes are.

**Comment by PJColombo** [comment]
Another request:

Can we move the `autocompleteBlockHash` function back to the [`optimism`](https://github.com/Blobscan/blobscan/blob/f21f6fcb752f6ce5f25339b426e1dcbb29d9d280/packages/api/src/blob-parse/optimism.ts) file?

**Comment by PJColombo** [comment]
A few more changes @luis-herasme  we should do the ui decoded fields based on new info learned about the blobs:

- If we successfully autocomplete the l1 origin block it means we have it on our own db so we can link to that block in blobscan instead of etherscan.
- Let's add a link to the "Last L1 origin number" and redirect to blobscan or etherscan based on previous described condition

File: apps/web/src/pages/tx/[hash].tsx [hunk 1]
```diff
@@ -281,30 +281,28 @@ const Tx: NextPage = () => {
                 },
                 {
                   name: "Parent L2 block hash",
-                  value: (
-                    <div className="flex items-center gap-2">
-                      {"0x" + decodedData.parentL2BlockHash + "..."}
-                    </div>
-                  ),
+                  value: "0x" + decodedData.parentL2BlockHash + "...",
```
Comment (xFJA) [inline apps/web/src/pages/tx/[hash].tsx:284]: I've found `Ox` several times repeated as a raw string. I would create a const for it

**Review by xFJA — CHANGES_REQUESTED** [review]
Missing changeset

File: apps/web/src/pages/tx/[hash].tsx (hunk 1 above)
Comment (PabloCastellano) [reply apps/web/src/pages/tx/[hash].tsx:284]: Beginning with 0x just means that the string is hexadecimal. I feel like having const for just a two characters string is kind of an overkill

**Review by PabloCastellano — APPROVED** [review]

**Comment by PabloCastellano** [comment]
@luis-herasme Can you also apply my changes from https://github.com/Blobscan/blobscan/pull/629 ?

**Review by xFJA — APPROVED** [review]

**Pull request merged**